qiskit/algorithms/time_evolvers/variational/solvers/var_qte_linear_solver.py

```diff
@@ -0,0 +1,123 @@
+        t_param: Parameter | None = None,
+        lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] | None = None,
+        imag_part_tol: float = 1e-7,
+    ) -> None:
+        """
+        Args:
+            var_principle: Variational Principle to be used.
+            hamiltonian: Operator used for Variational Quantum Time Evolution.
+            ansatz: Quantum state in the form of a parametrized quantum circuit.
+            gradient_params: List of parameters with respect to which gradients should be computed.
+                If ``None`` given, gradients w.r.t. all parameters will be computed.
+            t_param: Time parameter in case of a time-dependent Hamiltonian.
+            lse_solver: Linear system of equations solver callable. It accepts ``A`` and ``b`` to
+                solve ``Ax=b`` and returns ``x``. If ``None``, the default ``np.linalg.lstsq``
+                solver is used.
+            imag_part_tol: Allowed value of an imaginary part that can be neglected if no
+                imaginary part is expected.
+        """
+        self._var_principle = var_principle
+        self._hamiltonian = hamiltonian
+        self._ansatz = ansatz
+        self._gradient_params = gradient_params
+        self._bind_params = gradient_params
+        self._time_param = t_param
+        self.lse_solver = lse_solver
+        self._imag_part_tol = imag_part_tol
+
+    @property
+    def lse_solver(self) -> Callable[[np.ndarray, np.ndarray], np.ndarray]:
+        """Returns an LSE solver callable."""
+        return self._lse_solver
+
+    @lse_solver.setter
+    def lse_solver(self, lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] | None) -> None:
+        """Sets an LSE solver. Uses a ``np.linalg.lstsq`` callable if ``None`` provided."""
+        if lse_solver is None:
+            lse_solver = lambda a, b: np.linalg.lstsq(a, b, rcond=1e-2)[0]
+
+        self._lse_solver = lse_solver
+
+    def solve_lse(
+        self,
+        param_dict: dict[Parameter, float],
+        time_value: float | None = None,
+    ) -> (list | np.ndarray, list | np.ndarray, np.ndarray):
+        """
+        Solve the system of linear equations underlying McLachlan's variational principle for the
+        calculation without error bounds.
+
+        Args:
+            param_dict: Dictionary which relates parameter values to the parameters in the ansatz.
+            time_value: Time value that will be bound to ``t_param``. It is required if ``t_param``
+                is not ``None``.
+
+        Returns:
+            Solution to the LSE, A from Ax=b, b from Ax=b.
+
+        Raises:
+            ValueError: If time dependence expected from a non ``SparsePauliOp`` operator.
+
+        """
+        param_values = list(param_dict.values())
+        metric_tensor_lse_lhs = self._var_principle.metric_tensor(self._ansatz, param_values)
+        hamiltonian = self._hamiltonian
+
+        if self._time_param is not None:
```

Comment:
Above it states

> time_value: Time value that will be bound to ``t_param``. It is required if ``t_param`` is not ``None``.

Now time_value defaults to None but there seems to be no check here that is t_param (_time_param) is not None that time_value has indeed been passed. Would this presently end up failing somehow inside assign_parameters or maybe later where it would be more difficult to resolve. Could just default to some actual value? rather than None - or maybe that, while not failing, may also not be desirable?

Reply:
I decided to raise an error if no time value is provided. I am not sure about the desirability of providing a default here, and rather be on the safe side.